src/main/java/lavalink/client/io/LavalinkRestClient.java

```diff
@@ -0,0 +1,230 @@
+package lavalink.client.io;
+
+import com.sedmelluq.discord.lavaplayer.player.AudioLoadResultHandler;
+import com.sedmelluq.discord.lavaplayer.tools.FriendlyException;
+import com.sedmelluq.discord.lavaplayer.track.AudioPlaylist;
+import com.sedmelluq.discord.lavaplayer.track.AudioTrack;
+import com.sedmelluq.discord.lavaplayer.track.BasicAudioPlaylist;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import lavalink.client.LavalinkUtil;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+@SuppressWarnings("unused")
+public final class LavalinkRestClient {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LavalinkRestClient.class);
+
+    private static final Function<JSONObject, List<AudioTrack>> SEARCH_TRANSFORMER = loadResult -> {
+        final List<AudioTrack> tracks = new ArrayList<>();
+        final JSONArray trackData = loadResult.getJSONArray("tracks");
+
+        for (final Object track : trackData) {
+            try {
+                final AudioTrack audioTrack = LavalinkUtil.toAudioTrack(((JSONObject) track).getString("track"));
+                tracks.add(audioTrack);
+            } catch (final IOException exception) {
+                LOGGER.error("Error converting track", exception);
+            }
+        }
+
+        return tracks;
+    };
+
+    private final LavalinkSocket socket;
+    private Consumer<HttpClientBuilder> builderConsumer;
+
+    public LavalinkRestClient(final LavalinkSocket socket) {
```

Comment:
Should probably be private

Reply:
If you're talking about the constructor, that can't be private since it's used in the `Link` and `LavalinkSocket` classes (to register a new instance of it). It can have weaker access though, so maybe I should make it package-private

Reply:
Ah yes, I meant package-private